responsibleai/responsibleai/managers/data_balance_manager.py

```diff
@@ -0,0 +1,433 @@
+# Copyright (c) Microsoft Corporation
```

Comment:
I don't see any tests for the python and javascript code.  I think you will need to add a lot of tests for all of the new code.
I would recommend breaking this PR into python and typescript, and maybe get the python in first with all of the necessary tests implemented.